exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -404,21 +381,23 @@ public static Builder builder() {
 
     private EmulatedNodeType nodeType = EmulatedNodeType.VALIDATOR;
     private short validatorCount = 1;
-    private final List<Class<? extends ServiceModule>> services = new ArrayList<>();
-    private TimeProvider timeProvider;
+    private Multimap<ServiceArtifactId, ServiceSpec> services = ArrayListMultimap.create();
+    private HashMap<ServiceArtifactId, String> serviceArtifactFilenames = new HashMap<>();
+    private TimeServiceSpec timeServiceSpec;
```

Comment:
I think it must not be a single instance — time service, as any other, may be instantiated several times.

Reply:
Yes, I wasn't sure whether to allow several time services (why would anyone need that?), but I see no reason for us to restrict that. Fixed.

Reply:
I think we need it for testing purposes of our implementation (that it correctly supports multiple time services: #1136 ), but yes, I don't see either a realistic case when someone would need several time services in a permissioned system.